crates/rustc_codegen_spirv/src/linker/inline.rs

```diff
@@ -364,27 +386,28 @@ impl Inliner<'_, '_> {
         let call = caller.blocks[block_idx].instructions.pop().unwrap();
         assert!(call.class.opcode == Op::FunctionCall);
 
-        if let Some(call_result_type) = call_result_type {
-            // Generate the storage space for the return value: Do this *after* the split above,
-            // because if block_idx=0, inserting a variable here shifts call_index.
-            insert_opvariable(
-                &mut caller.blocks[0],
-                self.ptr_ty(call_result_type),
-                return_variable.unwrap(),
-            );
-        }
-
-        // Fuse the first block of the callee into the block of the caller. This is okay because
-        // it's illegal to branch to the first BB in a function.
-        let mut callee_header = inlined_blocks.remove(0).instructions;
+        // Move the variables over from the inlined function to here.
+        let mut callee_header = take(&mut inlined_blocks[0]).instructions;
         // TODO: OpLine handling
-        let num_variables = callee_header
-            .iter()
-            .position(|inst| inst.class.opcode != Op::Variable)
-            .unwrap_or(callee_header.len());
-        caller.blocks[block_idx]
-            .instructions
-            .append(&mut callee_header.split_off(num_variables));
+        let num_variables = callee_header.partition_point(|inst| inst.class.opcode == Op::Variable);
+        // Rather than fuse blocks, generate a new jump here. Branch fusing will take care of
+        // it, and we maintain the invariant that current block has finished processing.
```

Comment:
Clarifying this comment would be nice - "Rather than fuse blocks" is in reference to the old state of the code, and if someone doesn't know the old state, it's kind of vague. Maybe something like "Rather than fuse the first block of the inlined function into the current block of the caller, ..." or something like that.